tests/parser.test.js

```diff
@@ -0,0 +1,5288 @@
+import { beforeAll, describe, expect, it } from 'vitest';
+import { parseForESLint } from '../src/parser/gjs-gts-parser.js';
+import { traverse } from '../src/parser/transforms.js';
+import { SourceCode } from 'eslint';
+
+describe('transform', () => {
+  let text, result;
+  beforeAll(() => {
+    text = `
+    import { ExternalLink, Link, service } from 'limber-ui';
+
+const ReportIssue = <template>
+  If the tutorial navigated you here,
+  <ExternalLink href="https://github.com/NullVoxPopuli/limber/issues">please report the issue</ExternalLink>.
+  ❤️
+</template>;
+
+const CurrentPath = <template>
+  {{#let (service "docs") as |docs|}}
+    <code>{{docs.currentPath}}</code>
+  {{/let}}
+</template>;
+
+const BackToStart = <template>
+  You may also try going
+  <Link href="/" style="width: max-content; display: inline-block;">back to the beginning</Link>
+</template>;
+
+export const NotFound = <template>
+  Prose for the current tutorial, <CurrentPath />, could not be found.
+
+  Please check the URL and try again,
+  or navigate to a different tutorial chapter.
+
+  <br /><br />
+  <ReportIssue />
+
+  <br /><br /><br>
+  <BackToStart />
+</template>;
+    `;
+
+    result = parseForESLint(text, {
+      filePath: 'example.gts',
+      comment: true,
+      loc: true,
+      range: true,
+      tokens: true,
+    });
+  });
+
+  it('all tokens are correct', () => {
+    expect(
+      result.ast.tokens.map((t) => ({
+        value: t.value,
+        loc: t.loc,
+        range: t.range,
+      }))
+    ).toMatchInlineSnapshot(`
+      [
```

Comment:
Is there a way we can reduce this?

For asserting an internal structure, it seems sike something this big would be hard to work with/debug?

Reply:
tried some improvements